### ISSUE TITLE
Tag on Execute

### ISSUE THREAD

**Issue by mtekman** [galaxyproject/galaxy#8710]
It would be great to specify tags on dataset output when you click *Execute*

![Untitled](https://user-images.githubusercontent.com/20641402/65676897-f5f52500-e050-11e9-9b1d-bd430da6fc97.png)

**Comment by bernt-matthias** [comment]
Like the idea. Wondering if tags can be specified for workflow steps? If yes the suggestion would unify the behaviour. If no or should be available in both cases.

**Comment by mtekman** [comment]
Yeah in the workflow editor you can specify the output tags to add or remove for each tool.

**Comment by jmchilton** [comment]
https://github.com/galaxyproject/galaxy/pull/8948/files added the ability to use workflow post job actions directly on tool executions and show how to interact with the client to get this working.

**Comment by nomadscientist** [comment]
This looks ace! Also adding the ability to pre-label a dataset before pressing EXECUTE would be awesome. Such a faff waiting for it to finish, then hitting edit, then waiting for it to rename.

**Comment by mtekman** [comment]
**Another idea to add to the Execute dropdown box**

Upon re-running a tool, add a "Replace existing dataset?" option, very similar to the extra toggle that appears when a tools fails:

![image](https://user-images.githubusercontent.com/20641402/205068097-c6fe5603-27be-4e3a-b916-68d8a3289ebe.png)

But instead of being triggered on a failed job, it can be available when a tool is rerun, like so:

![justone](https://user-images.githubusercontent.com/20641402/205070924-e92375f3-d384-4a0c-8468-b9fb22c87ed5.png)

It could be similarly used to trigger the `job_remap: true` setting (in the POST response).

**Use cases**

* A user mistakenly runs a job with bad parameters, and would rather forget that it ever happened.
* A tool fails but is not caught by the tool error handler (e.g. due to problems with the server instead of the tool itself)

**Comment by wm75** [comment]
The big problem with that is that it could put your history into an inconsistent state if other jobs have used the first version of the dataset as input.

**Comment by mtekman** [comment]
Ah true...

I was going to suggest that the solution be that all downstream dependent datasets be marked as "failed" and then re-run when this dataset is renewed, similar to when a single dataset in a workflow fails... but I realise now that explicitly failed datasets have an important safety feature: they can't be used as inputs for further input. Datasets that were green before but red later don't have that safety.

Okay, idea aborted! :-)

**Comment by PlushZ** [comment]
I am currently working on this functionality (specifying tags on dataset output from Tool Form)
